src/main_nep/fitness.cu

```diff
@@ -218,13 +223,21 @@ void Fitness::output(
       for (int m = 0; m < dataset.Na_cpu[nc]; ++m) {
         data_nc += prediction[offset + m];
       }
-      fprintf(fid, "%g ", data_nc / dataset.Na_cpu[nc]);
+      if (!is_stress) {
+        fprintf(fid, "%g ", data_nc / dataset.Na_cpu[nc]);
+      } else {
+        fprintf(fid, "%g ", data_nc / dataset.structures[nc].volume * PRESSURE_UNIT_CONVERSION);
+      }
     }
     for (int n = 0; n < num_components; ++n) {
+      float ref_value = reference[n * dataset.Nc + nc];
+      if (is_stress) {
+        ref_value *= dataset.Na_cpu[nc] / dataset.structures[nc].volume * PRESSURE_UNIT_CONVERSION;
+      }
```

Comment:
It might be a bit cleaner to perform the conversion of the stress outside this function, e.g. before line 507. Something like (pseudocode)

```cpp
const vir = dataset.virial_cpu.data()
const vir_ref = dataset.virial_ref_cpu.data()
const stress = vir.copy()
const stress_ref = vir_ref.copy() 
for (int m...) {
    stress[m] *= data_nc / dataset.structures[nc].volume * PRESSURE_UNIT_CONVERSION
    stress_ref[m] *= data_nc / dataset.structures[nc].volume * PRESSURE_UNIT_CONVERSION
}

output(true, 6, fid_stress, stress, stress_ref, dataset);
```

That way, `Fitness::output` does not need to be changed and does not need the if statements, which would make it easier to read.

Reply:
But maybe this conversion would need to be done in multiple places, which would not be ideal...

Reply:
ometimes using a very large batchsize takes a lot of memory (there is such need for e.g., training a periodic table one) and I have tried to minimze memory usage, not to make a copy if it can be avoided.